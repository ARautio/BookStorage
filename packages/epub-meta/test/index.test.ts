import EPub from '../src/index';

const BOOK_FILE = './test/fixtures/pg2097-images.epub';

describe('epub', () => {
  describe('loadEpub', () => {
    it('should be able to load', async () => {
      const epub = new EPub(BOOK_FILE);
      await epub.load();
      expect(epub.metadata).toMatchSnapshot();
    });
  });
  describe('get cover', () => {
    it('should be able to load', async () => {
      const epub = new EPub(BOOK_FILE);
      await epub.load();
      const cover = await epub.getCover();
      expect(Buffer.isBuffer(cover.file)).toEqual(true);
    });
  });
});
