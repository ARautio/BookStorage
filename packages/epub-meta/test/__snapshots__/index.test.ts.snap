// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`epub loadEpub should be able to load 1`] = `
Object {
  "author": Array [
    "Arthur Conan Doyle",
  ],
  "coverPath": "@public@vhost@g@gutenberg@html@files@2097@2097-h@images@cover.jpg",
  "date": 2000-03-01T00:00:00.000Z,
  "isbn": undefined,
  "title": Array [
    "The Sign of the Four",
  ],
}
`;
